eslint/index.js

```diff
@@ -0,0 +1,57 @@
+module.exports = {
+  rules: {
+    'proptype-definition': {
+      create(context) {
+        return {
+          AssignmentExpression(node) {
+            // helper func which targets possible react components
+            const findDeclaration = singleNode => {
+              if (
+                singleNode.type.match(/[Function|Variable|Class|Export(Named|Default)]Declaration/)
+              ) {
+                // deal with default or named export
+                if (singleNode.declaration) {
+                  const { declarations } = singleNode.declaration;
+                  if (
+                    (declarations && declarations[0].id.name === propertyTypeNode.name) ||
+                    singleNode.declaration.id.name === propertyTypeNode.name
+                  ) {
+                    return true;
+                  }
+                  // deal with variable declaration or function statement
+                } else if (
+                  (singleNode.id && singleNode.id.name === propertyTypeNode.name) ||
+                  (singleNode.declarations &&
+                    singleNode.declarations[0].id.name === propertyTypeNode.name)
+                ) {
+                  return true;
+                }
+              }
+              return false;
+            };
+            // PropType node
+            const propertyTypeNode = node.left.object;
```

Comment:
You can disable the lint rule for "no abbreviations" here becuz it's an API we don't control.

```suggestion
            // delete line 32 comment
            const propTypeNode = node.left.object; // eslint-disable unicorn/prevent-abbreviations
```

Reply:
Will do.